cdap-api/src/main/java/co/cask/cdap/api/artifact/ArtifactVersion.java

```diff
@@ -30,11 +30,12 @@
  * parse version string in format of [major].[minor].[fix](-|.)[suffix]
  * }
  * </pre>
+ * Note that the suffix can not have numerical values in it.
  */
 @Beta
 public final class ArtifactVersion implements Comparable<ArtifactVersion> {
 
-  private static final Pattern PATTERN = Pattern.compile("(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:[.\\-](.*))?");
+  private static final Pattern PATTERN = Pattern.compile("(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:[.\\-]([^\\d]*))?$");
```

Comment:
The version could also be "1-foo-1", for instance.
Major version is "1", and suffix is "foo-1".